src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResource.java

```diff
@@ -125,27 +125,37 @@ public Response describeManageDataAccessRequests(@Auth AuthUser authUser) {
     @GET
     @Produces("application/json")
     @Path("/manage/v2")
-    @RolesAllowed({ADMIN, CHAIRPERSON, MEMBER, SIGNINGOFFICIAL})
+    @RolesAllowed({ADMIN, CHAIRPERSON, MEMBER, SIGNINGOFFICIAL, RESEARCHER})
     public Response describeManageDataAccessRequestsV2(@Auth AuthUser authUser, @QueryParam("roleName") Optional<String> roleName) {
         try {
             User user = userService.findUserByEmail(authUser.getName());
             String roleNameValue = roleName.orElse(null);
-            if (Objects.nonNull(roleNameValue)) {
-                boolean valid = UserRoles.isValidRole(roleNameValue);
-                if (valid) {
-                    //if the roleName is SO and the user does not have that role throw an exception
-                    if (roleNameValue.equalsIgnoreCase(UserRoles.SIGNINGOFFICIAL.getRoleName())) {
-                        if (!user.hasUserRole(UserRoles.SIGNINGOFFICIAL)) {
-                           throw new NotFoundException("User: " + user.getDisplayName() + ", " + " does not have Signing Official role.");
-                       }
-                   } else {
-                        throw new BadRequestException("Signing Official Role is the only role supported at this time");
-                    }
-                } else {
+            UserRoles queriedUserRole = UserRoles.getUserRoleFromName(roleNameValue);
+            if (roleName.isPresent()) {
+                //if a roleName was passed in but it is not in the UserRoles enum throw exception
+                if (Objects.isNull(queriedUserRole)) {
                     throw new BadRequestException("Invalid role name: " + roleNameValue);
+                } else {
+                    //if there is a valid roleName but it is not SO or Researcher then throw an exception
+                    if (queriedUserRole != UserRoles.RESEARCHER && queriedUserRole != UserRoles.SIGNINGOFFICIAL) {
+                        throw new BadRequestException("Unsupported role name: " + roleName);
+                    }
+                    //if the user does not have the given roleName throw NotFoundException
+                    if (!user.hasUserRole(UserRoles.getUserRoleFromName(roleNameValue))) {
+                        throw new NotFoundException("User: " + user.getDisplayName() + ", does not have " + roleName + " role.");
```

Comment:
This comes through as:
>   "message": "User: Gregory Rushton, does not have Optional[SigningOfficial] role.",

Due to `roleName` being an `Optional`. Should dereference it with `roleName.get()`
Same thing on line 141.

Reply:
Even better, refer to the actual enum:
```
throw new BadRequestException("Unsupported role name: " + queriedUserRole.getRoleName());
```